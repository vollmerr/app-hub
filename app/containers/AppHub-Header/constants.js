/*
 *
 * AppHubHeader constants
 *
 */

export const DEFAULT_ACTION = 'app/AppHubHeader/DEFAULT_ACTION';
