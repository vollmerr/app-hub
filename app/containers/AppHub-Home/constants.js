/*
 *
 * AppHubHome constants
 *
 */

export const DEFAULT_ACTION = 'app/AppHubHome/DEFAULT_ACTION';
