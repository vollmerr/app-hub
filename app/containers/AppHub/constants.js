// VIEW
export const CHANGE_MOBILE = 'AppHub/CHANGE_MOBILE';
export const CHANGE_PANEL_OPEN = 'AppHub/CHANGE_PANEL_OPEN';
export const CHANGE_PANEL_SELECTED = 'AppHub/CHANGE_PANEL_SELECTED';
// panel names
export const HELP_PANEL = 'AppHub/HELP_PANEL';
export const APPS_PANEL = 'AppHub/APPS_PANEL';
export const APP_NAV_PANEL = 'AppHub/APP_NAV_PANEL';
export const ALERTS_PANEL = 'AppHub/ALERTS_PANEL';
export const DEV_PANEL = 'AppHub/DEV_PANEL';
// APP
export const CHANGE_APP = 'AppHub/CHANGE_APP';
export const CHANGE_APP_STATUS = 'AppHub/CHANGE_APP_STATUS';
// user
export const AUTH_USER = 'AppHub/AUTH_USER';
export const AUTH_USER_DONE = 'AppHub/AUTH_USER_DONE';
// REQUEST STATUS TYPES
export const REQUEST = '_REQUEST';
export const SUCCESS = '_SUCCESS';
export const FAILURE = '_FAILURE';

export const COL_TYPES = {
  DATE: 'DATE',
};

export const ROLES = {
  USER: 'AppHub User',
};

// D3
export const INCREMENT_RENDER_COUNT = 'AppHub/INCREMENT_RENDER_COUNT';
