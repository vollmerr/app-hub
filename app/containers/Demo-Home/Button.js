import React from 'react';

const Button = ({ onClick, disabled, text }) => ( // eslint-disable-line
  <button
    onClick={onClick}
    disabled={disabled}
    style={{ background: disabled ? '#666' : '#333', color: '#fff', padding: '10px 15px', cursor: 'pointer' }}
  >{text}</button>
);

export default Button;
