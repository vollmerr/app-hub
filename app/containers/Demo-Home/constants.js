export const EXAMPLE_DATA_REQUEST = 'app/DemoHome/EXAMPLE_DATA_REQUEST';
export const EXAMPLE_DATA_SUCCESS = 'app/DemoHome/EXAMPLE_DATA_SUCCESS';
export const EXAMPLE_DATA_FAILURE = 'app/DemoHome/EXAMPLE_DATA_FAILURE';
