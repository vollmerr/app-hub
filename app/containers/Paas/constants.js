export const GET_MANAGER_DATA_REQUEST = 'app/Paas/GET_MANAGER_DATA_REQUEST';
export const GET_MANAGER_DATA_SUCCESS = 'app/Paas/GET_MANAGER_DATA_SUCCESS';
export const GET_MANAGER_DATA_FAILURE = 'app/Paas/GET_MANAGER_DATA_FAILURE';

export const UPDATE_USERS_REQUEST = 'app/Paas/UPDATE_USERS_REQUEST';
export const UPDATE_USERS_SUCCESS = 'app/Paas/UPDATE_USERS_SUCCESS';
export const UPDATE_USERS_FAILURE = 'app/Paas/UPDATE_USERS_FAILURE';

export const GET_REPORT_DATA_REQUEST = 'app/Paas/GET_REPORT_DATA_REQUEST';
export const GET_REPORT_DATA_SUCCESS = 'app/Paas/GET_REPORT_DATA_SUCCESS';
export const GET_REPORT_DATA_FAILURE = 'app/Paas/GET_REPORT_DATA_FAILURE';

export const SET_REPORT_KEY = 'app/Paas/SET_REPORT_KEY';
export const SET_REPORT_FILTER = 'app/Paas/SET_REPORT_FILTER';

// ROLES
export const ROLES = {
  MANAGER: 'PAAS Manager',
  SECURITY: 'PAAS Security',
  HR: 'PAAS HR',
};
export const ROLES_VALUES = Object.values(ROLES);
// APPROVAL CODES
export const APPROVAL = {
  NONE: undefined,
  DENY: 0,
  APPROVE: 1,
};
// ACCESS AUTHORIZATION
export const AUTH = {
  ID: '_id',
  SID: 'sid',
  FULL_NAME: 'fullName',
  EMAIL: 'email',
  POS_NUMBER: 'posNumber',
  MANAGER_SID: 'managerSid',
  MANAGER_NAME: 'managerFullName',
  STATUS: 'status',
  LAST_MODIFIED: 'lastUpdated',
  LAST_APPROVED: 'lastApproved',
  DATE_CREATED: 'created',
  AUTH_YEAR: 'authYear',
};
// AUTH STATUS
export const STATUS = {
  ACTIVE: 'active',
  INACTIVE: 'inactive',
  NO_MANAGER: 'noManager',
};
// APP TO AUTHORIZE
export const APPS = {
  APP_1: 'app1',
  APP_2: 'app2',
  APP_3: 'app3',
  APP_4: 'app4',
};
export const APP_LIST = Object.values(APPS);
export const BUTTONS = {
  APPROVE: 'approveButton',
  DENY: 'denyButton',
};
export const REPORT = {
  APPROVED: 0,
  DENIED: 1,
  PENDING: 2,
  NO_MANAGER: 3,
};
