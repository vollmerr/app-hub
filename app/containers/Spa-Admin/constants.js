export const FIELD_ACK_NAME = 'name';
export const FIELD_ACK_FILE = 'file';
