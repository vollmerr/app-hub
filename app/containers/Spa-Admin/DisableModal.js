import React from 'react';
import PropTypes from 'prop-types';
import { DefaultButton, PrimaryButton } from 'office-ui-fabric-react/lib/Button';
import { Dialog, DialogType, DialogFooter } from 'office-ui-fabric-react/lib/Dialog';


/**
 * Modal for confirming that a acknowledgment should be disabled
 */
class DisableModal extends React.PureComponent {
  render() {
    const {
      item,
      hidden,
      onClose,
      onConfirm,
    } = this.props;

    const { name, details } = item;

    const dialogProps = {
      hidden,
      onDismiss: onClose,
      dialogContentProps: {
        type: DialogType.normal,
        title: name,
        subText: details,
      },
    };

    const primaryButtonProps = {
      onClick: onConfirm,
      text: 'Confirm',
    };

    const cancelButtonProps = {
      onClick: onClose,
      text: 'Cancel',
    };

    return (
      <Dialog {...dialogProps}>
        <p>Are you sure you want to disable ... etc</p>
        <DialogFooter>
          <PrimaryButton {...primaryButtonProps} />
          <DefaultButton {...cancelButtonProps} />
        </DialogFooter>
      </Dialog>
    );
  }
}


const { bool, func, string, shape } = PropTypes;

DisableModal.propTypes = {
  item: shape({
    name: string,
    details: string,
  }),
  onConfirm: func,
  onClose: func,
  hidden: bool,
};

export default DisableModal;
