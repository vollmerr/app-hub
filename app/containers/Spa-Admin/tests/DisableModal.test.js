import React from 'react';
import { shallow } from 'enzyme';
import { DefaultButton, PrimaryButton } from 'office-ui-fabric-react/lib/Button';
import { Dialog, DialogFooter } from 'office-ui-fabric-react/lib/Dialog';

import { testStyledComponent } from 'utils/testUtils';
import { ACK } from 'containers/Spa/constants';

import DisableModal, { Warning } from '../DisableModal';

testStyledComponent(Warning);

const props = {
  type: 'test type',
  item: {
    [ACK.TITLE]: 'test title',
    [ACK.DETAILS]: 'test details',
  },
  onSubmit: jest.fn(),
  onClose: jest.fn(),
  hideModal: false,
};


describe('<DisableModal />', () => {
  let wrapper;
  beforeEach(() => {
    jest.resetAllMocks();
    wrapper = shallow(<DisableModal {...props} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a `Dialog`', () => {
    expect(wrapper.find(Dialog).length).toEqual(1);
  });

  it('should render some warning text', () => {
    expect(wrapper.find(Warning).length).toEqual(1);
  });

  it('should render a `DialogFooter` to contain action buttons', () => {
    expect(wrapper.find(Dialog).find(DialogFooter).length).toEqual(1);
  });

  it('should render a `PrimaryButton` for primary actions', () => {
    expect(wrapper.find(DialogFooter).find(PrimaryButton).length).toEqual(1);
  });

  it('should render a `DefaultButton` for secondary actions', () => {
    expect(wrapper.find(DialogFooter).find(DefaultButton).length).toEqual(1);
  });

  it('should `onSubmit` when clicking the primary button', () => {
    wrapper.find(PrimaryButton).simulate('click');
    expect(props.onSubmit).toHaveBeenCalled();
  });

  it('should `onClose` when clicking the secondary button', () => {
    wrapper.find(DefaultButton).simulate('click');
    expect(props.onClose).toHaveBeenCalled();
  });
});
