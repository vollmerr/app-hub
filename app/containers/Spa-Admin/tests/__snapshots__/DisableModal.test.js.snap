// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisableModal /> should render correctly 1`] = `
<WithResponsiveMode
  dialogContentProps={
    Object {
      "title": "test title",
      "type": 0,
    }
  }
  onDismiss={[Function]}
>
  <DisableModal__Warning>
    WARNING: This is no going back! Are you sure you want to test type test title?
  </DisableModal__Warning>
  <DialogFooter>
    <CustomizedPrimaryButton
      onClick={[Function]}
      text="Confirm"
    />
    <CustomizedDefaultButton
      onClick={[Function]}
      text="Cancel"
    />
  </DialogFooter>
</WithResponsiveMode>
`;

exports[`<DisableModal__Warning /> should render correctly 1`] = `
.c0 {
  font-weight: bold;
}

<p
  className="c0"
/>
`;
